packages/changed-elements-react/src/widgets/comparisonJobWidget/components/VersionCompareSelectModal.tsx

```diff
@@ -286,6 +275,33 @@ const createOrRunManagerStartComparisonV2 = async (args: RunStartComparisonV2Arg
   }
 };
 
+type handleJobErrorArgs = Omit<RunStartComparisonV2Args, "getDialogOpen" | "getToastsEnabled" | "runOnJobUpdate" | "iModelsClient"> & {
+  comparisonJob: ComparisonJob;
+};
+
+const handleJobError: (args: handleJobErrorArgs) => Promise<ComparisonJob> = async (args) => {
+  args.addPendingJob(args.comparisonJob.comparisonJob.jobId, {
+    targetNamedVersion: args.targetVersion,
+    currentNamedVersion: args.currentVersion,
+  });
+  await args.comparisonJobClient.deleteComparisonJob({
+    iTwinId: args.comparisonJob.comparisonJob.iTwinId,
+    iModelId: args.comparisonJob.comparisonJob.iModelId,
+    jobId: args.comparisonJob.comparisonJob.jobId,
+  });
+  return tryXTimes(async () => {
```

Comment:
You could check if Delete fails so that you don't run post 3 times when it won't work.

Reply:
If delete fails, an exception is thrown, and the caller of this function will catch it.